src/ngx_http_lua_socket_tcp.c

```diff
@@ -4908,65 +4901,30 @@ ngx_http_lua_socket_push_input_data(ngx_http_request_t *r,
     nbufs = 0;
     ll = NULL;
 
+    luaL_buffinit(L, &luabuf);
     for (cl = u->bufs_in; cl; cl = cl->next) {
+        ptrdiff_t chunk_size;
```

Comment:
@lneto Just look at the source code in the nginx core for both of these :) We just need to be consistent here.